keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,540 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.keyvault;
+
+import com.azure.common.ServiceClient;
+import com.azure.common.credentials.ServiceClientCredentials;
+import com.azure.common.http.HttpClient;
+import com.azure.common.http.HttpPipeline;
+import com.azure.common.http.policy.HttpLogDetailLevel;
+import com.azure.common.http.policy.HttpPipelinePolicy;
+import com.azure.common.http.policy.RetryPolicy;
+import com.azure.common.http.policy.UserAgentPolicy;
+import com.azure.common.http.policy.HttpLoggingPolicy;
+import com.azure.common.http.rest.RestResponse;
+import com.azure.common.implementation.RestProxy;
+import com.azure.keyvault.implementation.Page;
+import com.azure.keyvault.implementation.RestPagedResponseImpl;
+import com.azure.keyvault.models.DeletedSecret;
+import com.azure.keyvault.models.Secret;
+import com.azure.keyvault.models.SecretRequestAttributes;
+import com.azure.keyvault.models.SecretRequestOptions;
+import com.azure.keyvault.models.SecretBackup;
+import com.azure.keyvault.models.SecretInfo;
+import com.azure.keyvault.models.SecretRestoreRequestOptions;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.util.annotation.NonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+
+public final class SecretClient extends ServiceClient {
+    static final String SDK_NAME = "Azure-Keyvault";
+    static final String SDK_VERSION = "1.0.0-SNAPSHOT";
+    static final String API_VERSION = "7.0";
+    static final String ACCEPT_LANGUAGE = "en-US";
+    static final int DEFAULT_MAX_PAGE_RESULTS = 25;
+
+    private String vaultEndPoint;
+    private final SecretService service;
+
+
+    /**
+     * Creates a SecretClient that uses {@code credentials} to authorize with Azure and {@code pipeline} to
+     * service requests
+     *
+     * @param vaultEndPoint URL for the Application configuration service.
+     * @param pipeline HttpPipeline that the HTTP requests and responses flow through.
+     */
+    private SecretClient(String vaultEndPoint, HttpPipeline pipeline) {
+        super(pipeline);
+        this.vaultEndPoint = vaultEndPoint;
+        this.service = RestProxy.create(SecretService.class, this);
+    }
+
+    /**
+     * Creates a builder that can configure options for the SecretClient before creating an instance of it.
+     * @return A new Builder to create a SecretClient from.
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Provides configuration options for instances of {@link SecretClient}.
+     */
+    public static final class Builder {
```

Comment:
This should be called `SecretClientBuilder`

Reply:
Left it inside as Builder for now.

Reply:
Sync with @conniey on this - she went in a different direction after my review.